transform/set_slice_mangler.go

```diff
@@ -0,0 +1,65 @@
+package transform
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var (
+	emptyStructType  = reflect.TypeOf(struct{}{})
+	emptyStructValue = reflect.ValueOf(struct{}{})
```

Comment:
I think this is simpler as:

```suggestion
	emptyStructValue = reflect.ValueOf(struct{}{})
	emptyStructType  = emptyStructValue.Type()
```